src/common/compat.cc

```diff
@@ -16,6 +16,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <stdint.h>
+#include <stdio.h>
```

Comment:
i'd prefer `cstdio` over `stdio.h`. so all functions are in the namespace of `std`.

Reply:
> 
> 
> i'd prefer `cstdio` over `stdio.h`. so all functions are in the namespace of `std`.

Thusfar we have avoided using C++ libraries, which I really like.
Because there will come one day that uses a series of includes that will override `ENODATA`.  
And that is everytime a hard bug to detect and fix. So then perhaps enable the `#warning/#errors` in `src/include/compat.h`

Reply:
Tried doing that, but there a LOT of code files that include other stuff before `compat.h`, and thus define `ENODATA` to be 9919. Which comes out of the C++ includes.
So I guess that setting `#warnings/#errors` is a lost cause. Unless we want to overhaul just about every `.cc`-file header block.